crates/flox-rust-sdk/src/models/pkgdb_errors.rs

```diff
@@ -0,0 +1,109 @@
+use std::fmt::Display;
+
+use serde::Deserialize;
+use serde_json::Value;
+
+/// A struct representing error messages coming from pkgdb
+#[derive(Debug)]
+pub struct PkgDbError {
+    /// The exit code of pkgdb, can be used to programmatically determine
+    /// the category of error.
+    pub exit_code: u64,
+    /// The generic message for this category of error.
+    pub category_message: String,
+    /// The more contextual message for the specific error that occurred.
+    pub context_message: Option<ContextMsgError>,
```

Comment:
suggestion blocking: `category_message` and `context_message` are both optional (https://github.com/flox/pkgdb/blob/1fafafd9127fba3397e0c62d16545742abe442a5/include/flox/core/exceptions.hh#L101), and I don't think this struct can represent caught message existing but context message not. Add a comment saying we don't care about that state, or add a way to represent it

Reply:
Fixed